packages/datadog-instrumentations/src/http/client.js

```diff
@@ -39,6 +41,162 @@ function normalizeCallback (inputOptions, callback, inputURL) {
   return typeof inputOptions === 'function' ? [inputOptions, inputURL || {}] : [callback, inputOptions]
 }
 
+/**
+ * Wires the downstream response so we can observe when the customer consumes
+ * the body and when the stream finishes
+ *
+ * @param {object} ctx - Instrumentation context
+ * @param {import('http').IncomingMessage} res - The downstream response object.
+ * @returns {{ finalizeIfNeeded: () => void }|null} Cleanup helper used for drain.
+ */
+function setupResponseInstrumentation (ctx, res) {
+  const shouldInstrumentData = responseDataChannel.hasSubscribers
+  const shouldInstrumentFinish = responseFinishChannel.hasSubscribers
+
+  if (!shouldInstrumentData && !shouldInstrumentFinish) {
```

Comment:
Does this make sense if we are only subscribed to one of these? 

```suggestion
  if (!shouldInstrumentData || !shouldInstrumentFinish) {
```